src/TesApi.Web/BatchScheduler.cs

```diff
@@ -1724,22 +1326,67 @@ static bool TryGetValueAsDouble(Dictionary<string, string> dict, string key, out
                 .Select(line => { var parts = line.Split(fieldDelimiter); return new KeyValuePair<string, string>(parts[0], parts[1]); })
                 .ToDictionary(kv => kv.Key, kv => kv.Value);
 
+
+        /// <inheritdoc/>
+        public async IAsyncEnumerable<TesEventMessage> GetEventMessagesAsync([System.Runtime.CompilerServices.EnumeratorCancellation] CancellationToken cancellationToken, string @event)
```

Comment:
A more suitable place for this method is the TestEventMessage (RunnerEventsConsumer) class, not this class. There is no need to create a new instance of BatchScheduler to get messages.

Reply:
We don't create a new instance of `BatchScheduler`. We use the singleton from `Startup` that DI gives us.